### PR TITLE
Add C++ export (dependent on appleseed #2299)

### PR DESCRIPTION
**Pull request by jdent02**

### CODE REVIEW

File: translators/group.py [hunk 1]
```diff
@@ -108,7 +108,7 @@ def create_entities(self, scene):
 
         self.__assembly = asr.Assembly(self.assembly_name)
 
-        self._create_translators(scene)
+        self._create_translators(scene, use_cpp_export=False, skip_triangulation=False)
```
Comment (dictoon) [inline translators/group.py]: Thanks for using named arguments for boolean parameters!

File: translators/mesh.py [hunk 2]
```diff
@@ -310,103 +307,63 @@ def __convert_mesh(self, me):
...
 
-        # Vertices
-        self.__mesh_object.reserve_vertices(len(me.vertices))
-
-        for v in me.vertices:
-            self.__mesh_object.push_vertex(asr.Vector3f(v.co[0], v.co[1], v.co[2]))
-
-        # Faces.
-        self.__mesh_object.reserve_triangles(len(me.polygons))
+        do_normals = True if self.bl_obj.data.appleseed.export_normals else False
```
Comment (dictoon) [inline translators/mesh.py]: If `self.bl_obj.data.appleseed.export_normals` is already a boolean, then this simplifies to

    do_normals = self.bl_obj.data.appleseed.export_normals

File: translators/mesh.py [hunk 3]
```diff
@@ -310,103 +307,63 @@ def __convert_mesh(self, me):
...
-                uv_index += 1
+            uv_textures = me.tessface_uv_textures
 
-                uv = uv_layer[f.loop_indices[2]].uv
-                self.__mesh_object.push_tex_coords(asr.Vector2f(uv[0], uv[1]))
-                tri.m_a2 = uv_index
-                uv_index += 1
+            for uv in uv_textures:
+                if uv.active_render:
+                    active_uv = uv
```
Comment (dictoon) [inline translators/mesh.py]: Break?

File: translators/mesh.py [hunk 4]
```diff
@@ -310,103 +307,63 @@ def __convert_mesh(self, me):
...
+
+        me.calc_tessface()
+
+        vertex_pointer = me.vertices[0].as_pointer()
+        tessface_pointer = me.tessfaces[0].as_pointer()
+        vertices_length = len(me.vertices)
+        tessface_length = len(me.polygons)
+
+        asr.convert_bl_vertex_pose(self.__mesh_object,
+                                    pose,
```
Comment (dictoon) [inline translators/mesh.py]: Fix indentation.

**Comment by jdent02** [comment]
Too much clutter in here.  Will create a new PR once the appleseed PR settles down.

**Pull request closed without merging**